src/trait_group.h

```diff
@@ -0,0 +1,202 @@
+#pragma once
+#ifndef TRAIT_GROUP_H
+#define TRAIT_GROUP_H
+
+#include "json.h"
+#include "mutation.h"
+#include "string.h"
+#include "string_id.h"
+
+typedef std::string Trait_group_tag;
+typedef std::vector<trait_id> Trait_list;
+
+namespace trait_group
+{
+
+/**
+ * Returns a randomized list of traits from the given trait group.
+ */
+Trait_list traits_from( const Trait_group_tag &gid );
+
+/**
+ * Check whether a particular trait is a member of the given group.
+ */
+bool group_contains_trait( const Trait_group_tag &gid, const trait_id &tid );
+
+/**
+ * Check whether the given group ID has been defined.
+ */
+bool group_is_defined( const Trait_group_tag &gid );
```

Comment:
This function, for example, can be provided via the `string_id` interface. Instead of `group_is_defined(somegroup)`, one can write `somegroup.is_valid()`.